### PR TITLE
More speedup (~30%)

### PR DESCRIPTION
**Pull request by tribal-tec**
- remove target hooks (slow cmake_parse_arguments), functionality
  provided by common_application() and common_library()
- remove *package* target (long and slow target list); futhermore,
  CPack is not subproject aware anyways

### CODE REVIEW

**Comment by tribal-tec** [comment]
FYI @rdumusc @eile

**Comment by rdumusc** [comment]
+1, nice cleanup.
Note that you did not remove the package target for subprojects as this one comes from CPack, included by CommonCPack (and CPackConfig), but the dput* targets.

File: CommonCPack.cmake [hunk 1]
```diff
@@ -2,6 +2,12 @@
 # Copyright (c) 2012-2013 Stefan Eilemann <eile@eyescale.ch>
 # Info: http://www.itk.org/Wiki/CMake:Component_Install_With_CPack
 
+# No support for superproject packaging
```
Comment (rdumusc) [inline CommonCPack.cmake]: subproject packaging. With this, the super-project can still be packaged.

**Comment by eile** [comment]
+1

File: CommonCPack.cmake (hunk 1 above)
Comment (tribal-tec) [inline CommonCPack.cmake]: True. If the superproject includes CommonCPack. Not the case for config.bbp.

**Pull request merged**